oktapam/resource_server_checkout_settings.go

```diff
@@ -31,13 +30,54 @@ type serverCheckoutSettingsResource struct {
 }
 
 type serverCheckoutSettingsResourceModel struct {
-	Id                        types.String `tfsdk:"id"`
-	ResourceGroup             string       `tfsdk:"resource_group"`
-	Project                   string       `tfsdk:"project"`
-	CheckoutRequired          bool         `tfsdk:"checkout_required"`
-	CheckoutDurationInSeconds types.Int32  `tfsdk:"checkout_duration_in_seconds"`
-	IncludeList               []string     `tfsdk:"include_list"`
-	ExcludeList               []string     `tfsdk:"exclude_list"`
+	Id            types.String `tfsdk:"id"`
+	ResourceGroup string       `tfsdk:"resource_group"`
+	Project       string       `tfsdk:"project"`
```

Comment:
Project, ResourceGroup, Id are part of the URL, whereas ResourceCheckoutSettings is $ref'd in the spec. I _think_ would could in theory generate everything from the spec, but that feels like it's filled with sharp edges.

Instead I'm wondering if we autogenerated tfsdk versions of some entities, we could have a simpler* code generator to the ToSDK/FromSDK methods/functions?

Reply:
We can generate code from openapi spec using [terraform-plugin-codegen-openapi](https://github.com/hashicorp/terraform-plugin-codegen-openapi). It works but wherever we don't have a parity in structure between api and tf schema it will require some more customization. TF plugin generator also allow customizing mapping between OAS spec -> TF schema (https://github.com/hashicorp/terraform-plugin-codegen-openapi).

Reply:
@sachinsaxena-okta I was playing with that originally, but their code generator is still pretty lacking. 1) It wants to take over the entire code generation process for the provider, and 2) it doesn't support polymorphism yet. 

It definitely seems like the best overall future solution, but yeah it requires parity.